docs/source/examples/learning-depolarizing-noise.myst

```diff
@@ -0,0 +1,208 @@
+    ops,
+    unitary,
+    InsertStrategy,
+)
+from mitiq import Executor, Observable, PauliString
+from mitiq.interface.mitiq_cirq import compute_density_matrix
+from mitiq.cdr import generate_training_circuits
+from mitiq.cdr._testing import random_x_z_cnot_circuit
+from mitiq.pec import execute_with_pec
+from mitiq.pec.representations.depolarizing import (
+    represent_operation_with_local_depolarizing_noise,
+)
+from mitiq.pec.representations.learning import (
+    biased_noise_loss_function,
+    learn_depolarizing_noise_parameter,
+)
+```
+
+Create a simple Rx-Rz-CNOT circuit using `mitiq.cdr._testing.random_x_z_cnot_circuit`. Here we use an (optional) seed for reproducibility.
+
+```{code-cell} ipython3
+circuit = random_x_z_cnot_circuit(
+    LineQubit.range(2), n_moments=5, random_state=np.random.RandomState(1)
+)
+print(circuit)
+```
+
+In this example, we will learn the strength of depolarizing noise of the `CNOT` gate in the circuit.
+
+```{code-cell} ipython3
+CNOT_ops = list(circuit.findall_operations_with_gate_type(CXPowGate))
+index = CNOT_ops[0][0]
+op = CNOT_ops[0][1]
+operations_to_learn = [Circuit(op)]
+```
+
+Specify the observable of interest.
+
+```{code-cell} ipython3
+observable = Observable(PauliString("XZ"), PauliString("YY"))
+```
+
+Define the ideal executor function for simulating the Clifford training circuits without noise. This will be used for comparison to the error-mitigated expectation values in the learning function, and for a final comparison with the mitigated and unmitigated values at the end of the workflow.
+
+```{code-cell} ipython3
+def ideal_execute(circ: Circuit) -> np.ndarray:
+    return compute_density_matrix(circ, noise_level=(0.0,))
+
+
+ideal_executor = Executor(ideal_execute)
+```
+
+Define the noisy executor, in this case for simulating depolarizing noise on the `CNOT` gate in each of the training circuits. The optimized value of the noise strength `epsilon` should be close to the value defined for this executor.
+
+```{code-cell} ipython3
+epsilon = 0.05
+
+
+def noisy_execute(circ: Circuit) -> np.ndarray:
+    noisy_circ = circ.copy()
+    qubits = op.qubits
+    for q in qubits:
+        noisy_circ.insert(
+            index + 1,
+            DepolarizingChannel(epsilon)(q),
+            strategy=InsertStrategy.EARLIEST,
+        )
+    return ideal_execute(noisy_circ)
+
+
+noisy_executor = Executor(noisy_execute)
+```
+
+Before calling the optimizer, let's plot the loss function that will be minimized in the learning routine, over a small range of noise strength values. The loss function calls {func}`execute_with_pec`, and we can optionally pass keyword arguments to set the number of PEC samples, among other options. Here we set a relatively small value of `num_samples` to obtain a reasonable execution time. However, we avoid using a number of PEC samples that is too small, as it can result in a large statistical error and ultimately cause the optimization process to fail.
```

Comment:
{func} link is broken. I think the function name needs to start with `.`? I'm not entirely sure.

Reply:
that fixed it, thanks!